SimPEG/base/pde_simulation.py

```diff
@@ -23,6 +23,10 @@ def __inner_mat_mul_op(M, u, v=None, adjoint=False):
                 u = u[:, None, :]
             if adjoint:
                 if u.ndim > 1 and u.shape[-1] > 1:
+                    if v.shape[-1] > 2:
+                        return M.T * (
+                            u[:, None, :] * v.reshape((u.shape[0], -1, u.shape[1]))
+                        ).sum(axis=2)
```

Comment:
Can you tell me what input shapes you were sending to this function to make this necessary? I thought it already supported multiple polarizations already.